src/classes/BDI_FieldMappingCustomMetadata.cls

```diff
@@ -0,0 +1,225 @@
+                    BDI_TargetFields targetFields = new BDI_TargetFields();
+                    targetFields.addTargetField(targetObject, targetField);
+                    targetFieldsBySourceField.put(sourceSObjectField, targetFields);
+                }
+            }
+        }
+        return targetFieldsBySourceField;
+    }
+
+    /*******************************************************************************************************
+    * @description return singleton instance of the class
+    */
+    public static BDI_FieldMappingCustomMetadata getInstance() {
+        if (fieldMappingInstance == null) {
+            fieldMappingInstance = new BDI_FieldMappingCustomMetadata();
+            fieldMappingInstance.getFieldMappingSetName();
+            fieldMappingInstance.retrieveCustMetadata();
+        }
+        return fieldMappingInstance;
+    }
+
+    /*******************************************************************************************************
+    * @description For the provided fields (in the form DIObject.developerfieldname) and target object, 
+    * return a map of DI field (key) to mapped field name on the destination record (value). 
+    *
+    * @param dataImportObjectName the bdi object we care about (Contact1, Contact2, Account1, etc)
+    * @param targetObjectName the underlying object the bdi object is (ie, Contact, Account, etc)
+    * @param dataImportFields the DataImport fields to consider
+    * @return Map<String, String> a map of DataImport fields to underlying object fields (eg, Donation_Amount__c => Amount)
+    *
+    * TODO: Implement this method
+    */
+
+    public Map<String,String> getFieldMap(String dataImportObjectName, String targetObjectName, List<String> dataImportFields) {
+        Map<String,String> targetFieldByDataImportField = new Map<String,String>();
+
+        Data_Import_Field_Mapping__mdt[] diFieldMappings = fieldMappingsByObjMappingLegacyName.get(dataImportObjectName);
+
+        if (diFieldMappings != null) {
+            for (Data_Import_Field_Mapping__mdt difm : diFieldMappings) {
+
+                targetFieldByDataImportField.put(difm.Source_Field_API_Name__c,difm.Target_Field_API_Name__c);
+            }
+        }
+
+        return targetFieldByDataImportField;
+    }
+
+    private String getFieldMappingSetName(){
+        if (fieldMappingSetName == null) {
+            //Retreive the Data Import Settings to determine the default field mapping set.
+            Data_Import_Settings__c dis = UTIL_CustomSettingsFacade.getDataImportSettings();
+
+            fieldMappingSetName = UTIL_Namespace.adjustNPSPNamespace(dis.Default_Data_Import_Field_Mapping_Set__c);
+        }
+
+        return fieldMappingSetName;
+    }
+
+    private void retrieveCustMetadata(){
+        //If the code is not in the npsp namespace, then the npsp field prefixes will need to be systematically removed.
+        Boolean removeNPSPNamespace = (UTIL_Namespace.getNamespace() != 'npsp');
```

Comment:
Rather than hardcode `npsp` as the string to look for, I think the objective here is to check for any current package namespace. Basically something like the following, which would avoid hardcoding the namespace, would give you the expected result without the hardcoded string.
`Boolean removeNPSPNamespace = String.IsNotEmpty(UTIL_Namespace.getNamespace());`